File: hare/app_types/connector_text.py
"""
Connector text types.

Port of: src/types/connectorText.ts
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass
class ConnectorTextBlock:
    type: str = "connector_text"
    text: str = ""
    connector_id: str = ""


def is_connector_text_block(block: Any) -> bool:
    """Check if a block is a connector text block."""
    if isinstance(block, dict):
        return block.get("type") == "connector_text"
    if isinstance(block, ConnectorTextBlock):
        return True
    return False

File: hare/frontend/src/types/connectorText.ts
export type ConnectorTextBlock = {
  type: 'connector_text'
  connector_text: string
}

export type ConnectorTextDelta = {
  type: 'connector_text_delta'
  connector_text: string
}

export function isConnectorTextBlock(
  value: unknown,
): value is ConnectorTextBlock {
  return Boolean(
    value &&
      typeof value === 'object' &&
      (value as { type?: unknown }).type === 'connector_text' &&
      typeof (value as { connector_text?: unknown }).connector_text === 'string',
  )
}
